add meson commands to the Makefile

rhatdan opened the pull request:
I am not familiar with meson, so I think this will make it easier to discover the commands with a Makefile workflow.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

mwperina reviewed (commented):
I'm not sure about this, our goal was to use meson for everything, so remaining targets in Makefile has been left there only because we don't have yet solution in meson, which would do the same thing with the same comfort

engelmi commented on Makefile:
```diff
@@ -1,17 +1,33 @@
+	meson compile -C builddir
+
+test: all
+	meson setup builddir
```
`setup` and `compile` can be removed from the `test` target since it is calling `all`.

engelmi commented on Makefile:
```diff
@@ -1,17 +1,33 @@
 
 ALL_SRC_FILES = 'find . -name "*.[ch]" ! -path "./src/libhirte/ini/ini.[ch]" ! -path "./src/libhirte/hashmap/*" ! -path "./src/*/test/**" ! -path "./builddir/**" -print0'
 
+all:
```
Minor: Maybe this can be renamed to `compile`?

engelmi commented on Makefile:
```diff
@@ -1,17 +1,33 @@
+
+install: all
+	meson install -C builddir --destdir "$(DESTDIR)"
+
```
What do you think of a `clean` target? For example, like this
```bash
rm -rf builddir
meson setup builddir
```

rhatdan replied:
Yes definitely

eli-schwartz replied:
Ninja has a clean target -- it cleans all built outputs, but doesn't remove the build directory entirely, so you still have a ninja file. Adding a Makefile target that purges the entire build directory seems confusingly different from what meson/ninja does -- and it's also different from what other build systems do.

e.g. autotools distinguishes between `make clean` (removes built outputs) and `make distclean` (removes generated builddir, Makefiles, etc.)

rhatdan committed:
add meson commands to the Makefile

I am not familiar with meson, so I think this will make it easier to
discover the commands with a Makefile workflow.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

ygalblum commented on Makefile:
```diff
@@ -1,17 +1,38 @@
+
+clean:
+	rm -rf builddir
+	meson setup builddir
```
Why do you need to run the `setup` command during `clean`? Won't it run for `build`?

Yarboa replied:
It is a chicken and egg, since builddir already deleted, 
Maybe using this 
`meson setup --wipe builddir`

Could be better,

engelmi reviewed (approved):
LGTM

mwperina reviewed (approved):
LGTM
